File: Python/Bits/divisionParDeux.py
# Division d'un nombre par deux
def divisionParDeux(n: int) -> int:
    """
    Description:
        Fonction qui divise un nombre par deux, grace à l'opérateur de décalage binaire à droite (>>).

    Paramètres:
        n: {int} -- Le nombre à diviser par deux.

    Retourne:
        {int} -- Le résultat de la division par deux.
    
    Exemple:
        >>> divisionParDeux(10)
        5

        00000000000000000000000000001010 représente 10, soit 2^3 + 0 + 2^1 + 0
        00000000000000000000000000000101 représente 5, soit 2^1 + 0 + 2^0
    """
    return n >> 1

File: Go/Bits/divisionParDeux.go
package main

import "fmt"

//Fonction qui divise un nombre par deux, grace à l'opérateur de décalage binaire à droite (>>).
func divisionParDeux(n int) int{
    return n >> 1
}

func main() {
    fmt.Println(divisionParDeux(10))
}
//Output: 5